common/src/lock.rs

```diff
@@ -0,0 +1,117 @@
+//! A module containing [`lock_api`]-based lock types that are or are not `Send + Sync`
+//! depending on whether the `threading` feature of this module is enabled.
+
+use std::fmt;
+use std::marker::PhantomData;
+use std::ops::Deref;
+
+use lock_api::{
+    MappedMutexGuard, MappedRwLockReadGuard, MappedRwLockWriteGuard, Mutex, MutexGuard,
+    RawMutex as RawMutex_, RwLock, RwLockReadGuard, RwLockUpgradableReadGuard, RwLockWriteGuard,
+};
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "threading")] {
+        pub use parking_lot::{RawMutex, RawRwLock};
+
+        pub use once_cell::sync::{Lazy, OnceCell};
+    } else {
+        mod cell_lock;
+        pub use cell_lock::{RawCellMutex as RawMutex, RawCellRwLock as RawRwLock};
+
+        pub use once_cell::unsync::{Lazy, OnceCell};
```

Comment:
is this also belong to lock?

Reply:
I think it makes sense there; `sync::OnceCell` at least has to lock when it's initializing

Reply:
And it was there when this was `cell.rs` as well